front/lib/api/data_source_view.ts

```diff
@@ -109,6 +109,14 @@ async function getContentNodesForManagedDataSourceView(
     });
 
     if (connectorsRes.isErr()) {
+      if (connectorsRes.error.type === "connector_rate_limit_error") {
+        return new Err(
+          new Error(
+            connectorsRes.error.message ??
+              "An error occurred while fetching the resources' content nodes."
+          )
+        );
+      }
```

Comment:
Why not doing this for any error ? connectorsRes.error.message will usually provide more useful information ?

Reply:
I asked myself the question and decided to not do it because they are usually not user friendly messages and not meant to be user facing. I simplified the code pointed here though!